core/common/src/main/java/alluxio/exception/status/AlluxioStatusException.java

```diff
@@ -205,6 +205,25 @@ public static AlluxioStatusException fromCheckedException(Throwable throwable) {
   }
 
   /**
+   * Converts unchecked throwables to Alluxio status exceptions. Checked throwables should not be
+   * passed to this method.
+   *
+   * @param throwable a throwable
+   * @return the converted {@link AlluxioStatusException}
+   */
+  public static AlluxioStatusException fromUncheckedException(Throwable throwable) {
+    try {
+      throw throwable;
+    } catch (IllegalArgumentException e) {
+      return new InvalidArgumentException(e);
```

Comment:
The convention is that we use unchecked exceptions for exception states that indicate bugs. Otherwise we use checked exceptions. `IllegalArgumentException` would usually mean that there is a bug in how we make an internal method call on the server. The user probably doesn't know anything about the internal method, so it's better to just give them an `InternalException`.

`InvalidArgumentException` is for when the user supplies an argument that doesn't make sense. We should validate user inputs as early as possible to detect any errors that deserve and `InvalidArgumentException`.